eth/beacon/enums/signature_domain.py

```diff
@@ -1,7 +1,7 @@
 from enum import IntEnum
 
 
-class BLSDomain(IntEnum):
+class SignatureDomain(IntEnum):
```

Comment:
@hwwhww how would you feel about creating just a single file at `eth/beacon/types/enum.py` and then collecting all enums in that single file. Given the simplicity of enums I feel we don't need one file per enum and keeping them all in one files slightly cleans up imports.

I'd also recommend we rename `eth/beacon/types` to `eth/beacon/typing` because that is more inline with the rest of our naming (e.g. we have a library `eth_typing` and we also have an `eth.typing` module with is like a gateway for things that eventually may be extracted into `eth_typing).

In the long run (read: when ETH 2.0 / Serenity is stable), I believe these types would move into the `eth_typing` library anyway to be reusable across many different projects.

Reply:
> How would you feel about creating just a single file at eth/beacon/types/enum.py and then collecting all enums in that single file.

sounds good to me.

> I'd also recommend we rename `eth/beacon/types` to `eth/beacon/typing` because that is more inline with the rest of our naming

I stole the folder name `types` from `go-ethereum`. 😂
The role is similar to `eth/rlp/` instead these data structures will be SSZ serialization objects after we finish `py-ssz`. `typing` might be confused with simple type definition from `eth/typing.py`? The "correct" folder name might be `ssz`, although it's not ready, yet.

> In the long run (read: when ETH 2.0 / Serenity is stable), I believe these types would move into the eth_typing library anyway to be reusable across many different projects.

Agreed! ✨
I just updated https://github.com/ethereum/py-evm/issues/1383 description.

Reply:
>might be confused with simple type definition

The `eth_typing` as well as it's little sister `eth.typing` aren't exclusively for `NewType` or *type aliases* if that's what you mean by "type definitions".

`eth_typing` holds all types that are generic enough to be useful for other Ethereum related Python libraries. These types are usually very shallow and contain no or only minimal logic. They often fall into the category of `Enum`, `NewType`, `NamedTuple`, `TypedDict` as well as type aliases to specific `Tuple` or `Union` of some types.

The only difference between `eth_typing` and `eth.typing` is that the latter isn't entirely proven to be useful outside of the scope of this repository and so they reside within the repository unless someones screams "Hey, why are we duplicating all these types in project x that also exist in `eth.typing`". Then, we would move them into the `eth_typing` library.

>The role is similar to `eth/rlp/` instead these data structures will be SSZ serialization objects after we finish `py-ssz`

Right, I missed that. It kinda opens up the broader question how does `eth.rlp` relate to this. I guess the fact that they are all RLP serialized and contain some specific functionality makes them not a good fit to go into the broader scope of  `eth_typing`. But apart from that, there may be valid arguments for restructuring our modules to be more like:

```
eth/
    typing/
        rlp/
           eth1
              accounts.py
              blocks.py
              ...
           beacon
              attestation_data.py
              attestation_records.py
              ...
        misc.py (everything that is currently in eth.typing.py)
```

But this opens up an even bigger code organization question. Right now, everything ETH 2.0 related is sitting shielded in `eth/beacon` and while that is great for keeping it isolated from the rest of the library (especially while you keep iterating on the spec heavily) it does make some things less ergonomic I think.

Looking at the following two imports one can easily see that the beacon chain is something that was added later to this library.

```
from eth.rlp.accounts import Account
from eth.beacon.rlp.attestation_data import AttestationData
```

It's just not symmetric. To make it symmetric (and treating `rlp` as a submodule of `typing`) it would either be.

```
from eth.eth1.typing.rlp.accounts import Account
from eth.beacon.typing.rlp.attestation_data import AttestationData
```

or

```
from eth.typing.rlp.eth1.accounts import Account
from eth.typing.rlp.beacon.attestation_data import AttestationData
```

But this opens up a much bigger box of what I initially meant to discuss here (sorry). I think we touched on this topic in https://github.com/ethereum/py-evm/issues/930 before. Maybe it would be a good idea to continue the discussion with this specific example at hand. I think that at least *long* term we want to get rid of the asymmetry to not make the beacon support standout alien. /cc @djrtwo @pipermerriam @carver

Reply:
In generally I'm :+1: on figuring this out but I'm "meh" :neutral_face: on taking any major action towards it at this time.  Namespacing the `eth1` data structures and generally the `eth1` *stuff* seems appropriate at some point but low priority.

Reply:
Yes, we don't have to address this now. I guess it would be nice to address this at least before any release that declares stable beacon support though.

Reply:
@cburgdorf I put `ValidatorStatusCode`, `ValidatorRegistryDeltaFlag`, and `SignatureDomain` in `eth/beacon/enums.py` now. (using plural for being aligned with `constants.py`)
I hope that we will use SSZ before the stable beacon chain release. 😊